examples/ex010_advection.jl

```diff
@@ -0,0 +1,88 @@
+# ---------------------------------------------------------------
+# advection example following Melvin, Staniforth and Thuburn
+# Q.J:R. Meteorol. Soc. 138: 1934-1947, Oct (2012)
+# ---------------------------------------------------------------
+using LFAToolkit
+using LinearAlgebra
+using Plots
+
+function kosloff_tal_ezer(α)
+    g(s) = asin(α * s) / asin(α)
+    g_prime(s) = α / (asin(α) * sqrt(1 - (α * s)^2))
+    g, g_prime
+end
+
+function hale_trefethen_strip(ρ)
+    τ = π / log(ρ)
+    d = 0.5 + 1 / (exp(τ * π) + 1)
+    π2 = π / 2
+    # Unscaled functions of u
+    g_u(u) = log(1 + exp(-τ * (π2 + u))) - log(1 + exp(-τ * (π2 - u))) + d * τ * u
+    g_prime_u(u) = 1 / (exp(τ * (π2 + u)) + 1) + 1 / (exp(τ * (π2 - u)) + 1) - d
+    # Normalizing factor and scaled functions of s
+    C = 1 / g_u(π / 2)
+    g(s) = C * g_u(asin(s))
+    g_prime(s) = -τ * C / sqrt(1 - s^2) * g_prime_u(asin(s))
+    g, g_prime
+end
+
+# setup
+mesh = Mesh1D(1.0)
+P = 11;
+Q = P;
+collocate = false
+mapping = kosloff_tal_ezer(0.98)
+mapping = sausage(29)
+mapping = hale_trefethen_strip(1.4)
```

Comment:
@jeremylt How should we parametrize this? Should we have a simple plain advection operator and a separate example that does mapping? Maybe a helper function for plotting instead of eagerly plotting?